apps/etterlatte-saksbehandling-ui/client/src/components/behandling/vilkaarsvurdering/Vilkaarsvurdering.tsx

```diff
@@ -33,20 +32,20 @@ export const Vilkaarsvurdering = () => {
   useEffect(() => {
     if (!behandlingId) throw new Error('Mangler behandlingsid')
     if (!vilkaarsvurdering) {
-      fetchVilkaarsvurdering(
-        behandlingId,
-        (vilkaarsvurdering) => dispatch(updateVilkaarsvurdering(vilkaarsvurdering)),
-        (error) => opprettHvisDenIkkeFinnes(error)
-      )
+      fetchVilkaarsvurdering(behandlingId, (vilkaarsvurdering) => {
+        if (vilkaarsvurdering == null) {
+          opprettHvisDenIkkeFinnes()
+        } else {
+          dispatch(updateVilkaarsvurdering(vilkaarsvurdering))
+        }
+      })
     }
   }, [behandlingId])
 
-  const opprettHvisDenIkkeFinnes = (error: ApiError) => {
-    if (error.statusCode === 404) {
-      opprettNyVilkaarsvurdering(behandlingId!!, (vilkaarsvurdering) =>
-        dispatch(updateVilkaarsvurdering(vilkaarsvurdering))
-      )
-    }
+  const opprettHvisDenIkkeFinnes = () => {
+    opprettNyVilkaarsvurdering(behandlingId!!, (vilkaarsvurdering) =>
```

Comment:
Enten send inn behandlingId som non-nullable eller bare flytt disse to linjene inn i linje 37, så slipper vi `!!` 🤠

Reply:
c0c3fcff71511d487192c0bae23f8b26449ffb8d